fault-tolerance/src/main/java/io/helidon/faulttolerance/AsyncImpl.java

```diff
@@ -35,14 +36,17 @@
         CompletableFuture<T> future = new CompletableFuture<>();
         AsyncTask<T> task = new AsyncTask<>(supplier, future);
 
+        Future<?> taskFuture;
         try {
-            executor.get().submit(task);
+            taskFuture = executor.get().submit(task);
         } catch (Throwable e) {
             // rejected execution and other executor related issues
             return Single.error(e);
         }
 
-        return Single.create(future);
+        Single<T> single = Single.create(future, true);
+        single.onCancel(() -> taskFuture.cancel(false));     // cancel task
+        return single;
```

Comment:
This looks like duplication of `CompletableFuture.supplyAsync` internals, is there some `AsyncTask`'s feature I missed @tomas-langer, or we can simplify?
```java
    public <T> Single<T> invoke(Supplier<T> supplier) {
        try {
            return Single.create(CompletableFuture.supplyAsync(task, executor.get()), true);
        } catch (Throwable e) {
            // rejected execution and other executor related issues
            return Single.error(e);
        }
    }
```

Reply:
Maybe, but not sure it is more readable tbh

Reply:
Code has changed now, see below